### ISSUE TITLE
WAVE reports Skipped Heading Level alert in Funnel 2-up widget.

### ISSUE THREAD

**Issue by tatwell** [chapmanu/cascade-assets#249]
Trello: https://trello.com/c/gt9YZaT3

**Comment by tatwell** [comment]
Sample page: https://www.chapman.edu/admission/index.aspx

![screen shot 2017-09-28 at 8 38 26 am](https://user-images.githubusercontent.com/12798918/30975986-6f4842f0-a428-11e7-8998-7df41fbc59e8.png)

Heading should be `h2` not `h3`. 

Here's the markup (it's a direct child of `div#main`):

```
<div class="funnel-boxes-widget funnel">
  <div class="funnelBlock">
    <div class="featureImage">
      <a href="undergraduate/index.aspx"><img alt="Undergraduate Students" height="125px" src="_files/undergraduate-students-admission-main-page.jpg" style="" width="290px"><img alt="FEATURES: Linked image with alternative text" class="wave5icon" src="chrome-extension://jbbplnpkjmmeebjpijfedlgcdilocofh/img/icons/alt_link.png" style="border: none; vertical-align: bottom; cursor: pointer; max-width: 44px; max-height: 25px; z-index: 10000; display: inline;" tabindex="0"><span class="wave5text" style="display: inline; background-color: rgb(94, 140, 78); color: rgb(255, 255, 255); border: 1px solid rgb(204, 204, 204); height: auto; width: auto; z-index: 10; font-size: 14px; line-height: 20px; font-weight: normal; font-family: Helvetica, Arial, sans-serif; margin: 0px; padding: 0px;">*Undergraduate Students*</span></a>
    </div>
    <h3 style=""><img alt="STRUCTURAL ELEMENTS: Heading level 3" class="wave5icon" src="chrome-extension://jbbplnpkjmmeebjpijfedlgcdilocofh/img/icons/h3.png" style="border: none; vertical-align: bottom; cursor: pointer; max-width: 44px; max-height: 25px; z-index: 10000; display: inline;" tabindex="0"><img alt="ALERTS: Skipped heading level" class="wave5icon" src="chrome-extension://jbbplnpkjmmeebjpijfedlgcdilocofh/img/icons/heading_skipped.png" style="border: none; vertical-align: bottom; cursor: pointer; max-width: 44px; max-height: 25px; z-index: 10000; display: inline;" tabindex="0"> <a href="undergraduate/index.aspx" style="">Undergraduate Admission&nbsp;»<img alt="ALERTS: Redundant link" class="wave5icon" src="chrome-extension://jbbplnpkjmmeebjpijfedlgcdilocofh/img/icons/link_redundant.png" style="border: none; vertical-align: bottom; cursor: pointer; max-width: 44px; max-height: 25px; z-index: 10000; display: inline;" tabindex="0"></a></h3>
    <div class="copy">
      If you are a current high school student or graduate, undergraduate transfer student, or looking to earn an additional bachelor's degree, click the link above to take the first steps to a distinguished higher education.
    </div>
    <div class="linkList" style="margin-left: 2em"><img alt="STRUCTURAL ELEMENTS: Unordered list" class="wave5icon" src="chrome-extension://jbbplnpkjmmeebjpijfedlgcdilocofh/img/icons/ul.png" style="border: none; vertical-align: bottom; cursor: pointer; max-width: 44px; max-height: 25px; z-index: 10000; display: inline;" tabindex="0"></div>
  </div>

  <!-- Additional funnelBlocks removed. -->
  
</div>
```

**Comment by tatwell** [comment]
@mandythomas Do you know if it's possible for this widget `div.funnel-boxes-widget` to be nested under under element or widget? Or is it fixed by the template to always be under `div#main`?

The reason I ask is I'm trying to figure out if there would ever be a scenario where it might be nested under another `h2` header.

**Comment by tatwell** [comment]
Fix has been push and deployed to Dev Cascade. Alert no longer present on sample page:

- https://dev-www.chapman.edu/admission/index.aspx

**Comment by mandythomas** [comment]
Sorry, late in replying. This widget would never be nested under another, widget changing it's heading level. Each widget is unique to itself and if it has a heading it is H2. The only exception, and not in this widget's case are those widgets that in addition to the items (eg collapsibles or tabs) there is an optional, overarching heading above the set of collapsibles. Then the heading is h2 and collapsibe's headings each are h3. If that heading is left empty (because optional) then the collapsibles are each h2's. Very easy in the format to write it because the heading and collapsibles are all in the same widget and same format so in that case we would have tested for headline field to be blank and use if/else to surround headings in h2 or h3 tags as appropriate. Doesn't apply in this funnel 2-up widget though (currently).

**Comment by tatwell** [comment]
@mandythomas Thanks for the reply. It seems to me the safe approach for widgets that could appear at different levels of the page hierarchy -- and maybe widgets generally -- would be to avoid `h` tags altogether.

**Comment by mandythomas** [comment]
I have to say I do not agree. Appropriate Hn tags are the only way to make a page user friendly for a disabled user. If we remove them to make programming easy for us, we make the visitor read the entire text just to find out if the info they want, 5 paragraphs down is there. Headings allow them to scan the major topics on the page and land on the page, saving them time. It's hard enough being a blind user trying to glean info from a long webpage. The widgets will never change their headings depending on other widgets on the page. The headings are all self-contained within a widget. Each section (widget) is considered a major topic of content and will always begin with an H2. It's only when that widget had the ability to have additional levels or an optional heading that we get into the if/else but again it's all self contained within that one widget and it's velocity format. We do not have to check anything else on the page to make the Hn decision for that widget.

**Comment by tatwell** [comment]
> Each section (widget) is considered a major topic of content and will always begin with an H2.

If this rule is known and enforced, then it won't be an issue. 

If there exists widgets with structure:

```
<div class="widget">
  <h2>Header</h2>
  <div>Ipsum lorem...</div>
</div>
```

And it were nested in another widget or block like so:

```
<div class="some-macro">
  <h2>Header</h2>
  <div>Ipsum lorem...</div>
  
  #insertWidget()
</div>
```

I'd expect that would be a violation.

When you say:

> It's only when that widget had the ability to have additional levels...

I take that to mean a scenario like this might be possible. Granted, I don't know of any specific cases. 

Still, the fact that we're fixing this issue here indicates to me that we may not have the foresight or discipline to avoid these accessibility technicalities. In which case, given our limited resources and the institutional focus on full compliance, I think we'd be better served to err on the side of technical compliance than actual usability.

That said, until I encounter an actual exception, I'll assume this rule is absolute:

> Each section (widget) is considered a major topic of content and will always begin with an H2.

**Comment by mandythomas** [comment]
To clarify, users cannot insert one widget inside another. The templates aren't set up to allow that. But, it's not a mistake that we provided an optional heading in some widgets, it's a flexibility for the client. I think it's totally reasonable, and what Ross requested,  and we are just making sure we output appropriate html markup depending on the choice they make on the page. We just didn't get it right in our initial coding of this widget so that's what this fix is about. 

Here's an example if it helps. This page is a typical use, with no Heading used above the Collapsibles widget because it was a simple page and the page title said it all. https://www.chapman.edu/campus-services/information-systems/software/skype-for-business.aspx. 

However on this page https://www.chapman.edu/campus-services/information-systems/software/mathematica.aspx there are 2 independent sets of Collapsible widgets (the fact that they are the same widget is irrelevant). To distinguish one set from another the author used the optional Heading field within each widget to give the set of collapsibles some context. In velocity each widget is processed independently of whatever other widgets are on the page. The H2 vs H3 only comes into play WITHIN each widget if the user chose to supply an optional Heading for that widget. The Heading is an H2 if provided, and Collapsibles are then H3's. If no heading, then the Collapsibles are H2's. It's just the nesting within the widget that's important. This would be true even if the author put only one set of the Collapsible widget on the page and chose to provide a Heading or no Heading. Same logic still applies

**Comment by tatwell** [comment]
Format released to production. Republish admission index page and verified issue resolved:

- https://www.chapman.edu/admission/index.aspx
